src/App.js

```diff
@@ -29,37 +30,103 @@ const App = () => {
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [currentPlayer, setCurrentPlayer] = useState(PLAYER_1);
+  // const [winner, setWinner] = useState(null);
+  // const [gameEnd, setGameEnd] = useState(false);
 
-  // Wave 2
+  const squareClicked =(id)=>{
+    let didPlay = false 
+    if (winner !== null) {return}
```

Comment:
👍 
If winner has a value, then the game is over, and we shouldn't process any more board clicks. So we can simply return to exit the function without modifying any state.